pvt/benchmarks/src/PoolBenchmark.behavior.ts

```diff
@@ -651,105 +850,209 @@ export class Benchmark {
     });
 
-    describe('initialization', () => {
-      sharedBeforeEach(async () => {
-        poolTokens = sortAddresses([tokenAAddress, wethAddress]);
-        this.tokenConfig = buildTokenConfig(poolTokens, false);
-      });
-
-      itTestsInitialize(false);
-      itTestsInitialize(true);
-    });
-
-    describe('test yield pool', () => {
-      sharedBeforeEach(async () => {
-        poolTokens = sortAddresses([tokenAAddress, tokenBAddress]);
-        this.tokenConfig = buildTokenConfig(poolTokens, true);
-        await deployAndInitializePool();
+    describe('test yield pool', async () => {
+      sharedBeforeEach(`deploy pool`, async () => {
+        await cleanPools();
+        await deployAndInitializePool(PoolTag.WithRate, sortAddresses([tokenAAddress, tokenBAddress]), true);
         await this.tokenA.setRate(fp(1.1));
         await this.tokenB.setRate(fp(1.1));
       });
 
       context('swap', () => {
-        itTestsSwap('With rate', async () => {
-          await this.tokenA.setRate(fp(1.2));
-          await this.tokenB.setRate(fp(1.2));
+        itTestsSwap(PoolTag.WithRate, {
+          actionAfterFirstTx: async () => {
+            await this.tokenA.setRate(fp(1.2));
+            await this.tokenB.setRate(fp(1.2));
+          },
         });
       });
 
       context('remove liquidity', async () => {
-        itTestsRemoveLiquidity('With rate', async () => {
+        itTestsRemoveLiquidity(PoolTag.WithRate, async () => {
           await this.tokenA.setRate(fp(1.2));
           await this.tokenB.setRate(fp(1.2));
         });
       });
 
       context('add liquidity', async () => {
-        itTestsAddLiquidity('With rate', async () => {
-          await this.tokenA.setRate(fp(1.2));
-          await this.tokenB.setRate(fp(1.2));
+        itTestsAddLiquidity(PoolTag.WithRate, {
+          actionAfterFirstTx: async () => {
+            await this.tokenA.setRate(fp(1.2));
+            await this.tokenB.setRate(fp(1.2));
+          },
         });
       });
     });
 
-    describe('test donation', () => {
-      itTestsDonation();
+    describe('test nested pool', async () => {
+      sharedBeforeEach(`deploy pool`, async () => {
+        await cleanPools();
+        await deployAndInitializePool(PoolTag.Nested, sortAddresses([tokenAAddress, tokenBAddress]), false);
+
+        const pool = await this.poolsInfo[PoolTag.Nested].pool;
+
+        const bptBalance = await this.vault.balanceOf(pool, alice);
+
+        await (pool as IERC20).connect(alice).approve(permit2, MAX_UINT256);
+        await permit2.connect(alice).approve(pool, router, MAX_UINT160, MAX_UINT48);
+        await permit2.connect(alice).approve(pool, batchRouter, MAX_UINT160, MAX_UINT48);
+
+        await deployAndInitializePool(PoolTag.WithBPT, sortAddresses([await pool.getAddress(), tokenCAddress]), false, [
+          bptBalance,
+          bptBalance,
+        ]);
+      });
+
+      context('swap', () => {
+        itTestsSwap(PoolTag.Nested, {
```

Comment:
Note that we want something a bit different from this one. I believe what we're measuring here is a swap internal to the parent pool; we also need the measurement end to end.

In other words, suppose we have:
- Pool A: waUsdc / waUsdt / waDai
- Pool B: Pool A / GHO

If I'm not mistaken this test measures a swap of Pool A for GHO right?

We also want to test DAI / GHO.
We must use the batch router for this, and we need initialized buffers (with enough liquidity). The batch router will figure all of this out given the correct swap path.

The swap path should be:
- DAI - waDAI (buffer wrap)
- waDAI - pool A (add liquidity)
- pool A - GHO (swap)

And also vice versa, GHO to DAI:
- GHO - Pool A (swap)
- Pool A - waDAI (exit)
- waDAI - DAI (unwrap)

Again, the batch router should figure everything out as long as the swap path is correctly set.